stdlib/public/core/Indices.swift.gyb

```diff
@@ -120,9 +120,6 @@ public struct ${Self}<
 }
 
 extension ${collectionForTraversal(Traversal)}
-%     if Traversal != 'RandomAccess':
-  where Indices == ${Self}<Self>
```

Comment:
Only the `%      if ...` should go away here, right? We want this extension to be constrained to where `Self.Indices == DefaultIndices<Self>` for each specialized version of `Default____Indices`.

Reply:
d'oh. right.

Reply:
I'm going to do this part in a separate PR along with another change.